### ISSUE TITLE
Support `get_file_slices_incremental` for table

### ISSUE THREAD

**Issue by jonathanc-n** [apache/hudi-rs#219]
### Description of the improvement

To support incremental queries:
- Create function called `get_file_slices_incremental` for table.
- The function should be given a `start_timestamp`  and a `Option<end_timestamp>` as the range; if end timestamp is `None`, then it will default to the current timestamp (latest commit instant). (partition pruner and excluded file groups will be included in parameters)
- Based on the `start_timestamp` and `end_timestamp` it will read when the most recent changes happened to each files, and only retrieve slice if it falls in the range.

### Expected behavior

Be able to return fileslices (only supporting basefiles currently; CoW) that have changes that fall in a range of one commit time to an optional timestamp (which is the current timestamp).

**Comment by jonathanc-n** [comment]
@xushiyan Let me know if this sounds fine, and assign; i will work on it

**Comment by xushiyan** [comment]
we need to finalize https://github.com/apache/hudi-rs/pull/203 first before moving onto incr query, which requires filtering on `_hoodie_commit_time` field, so we need to support generic filters properly.

**Comment by xushiyan** [comment]
@jonathanc-n i have merged the api stubs in #220 which you can implement incremental query flow. For the first impl, you don't need to handle archived timeline and non-latest inflight commits - those should be addressed in separate PRs.

**Comment by xushiyan** [comment]
hi @jonathanc-n any update on the work here?

**Comment by jonathanc-n** [comment]
@xushiyan Hello, I will have some time tomorrow, so I can prob put in a pull request for this.

**Comment by xushiyan** [comment]
hey @jonathanc-n i'll start on supporting archived timeline now and then move on to this main impl. If you have any WIP PR, feel free to put it up. It's totally fine if you didn't get chance for this. If you happen to have time in the next week or so, it'll be more helpful if you can take up the IN/NOT IN partition filter support work for the upcoming release.

**Comment by jonathanc-n** [comment]
Yeah I didn't foresee what was needed, i'll try to get started on the partition filter support.

**Comment by xushiyan** [comment]
closing this as it's covered in #9
